integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/EmployeePortalTestCase.java

```diff
@@ -94,4 +140,234 @@ public void cleanup() {
         } catch (Exception ignored) {
         }
     }
+
+    /**
+     * Get client id and client secret needed to generate oauth token to access apim.
+     *
+     * @return a concatenated string comprising client id and client secret delimited by a colon
+     * @throws NoSuchAlgorithmException if sendPost method fails
+     * @throws IOException              if sendPost method fails
+     * @throws KeyManagementException   if sendPost method fails
+     */
+    private String getClientIdColonClientSecret() throws NoSuchAlgorithmException
```

Comment:
Let's remove the 'colon' from the method name - getBased64EncodedClientIdSecret. And I think you have to convert it into Base64 format right? In that case, you could directly do that here doing that in the next method getOauthToken.

Reply:
Also, I think this method could move to utils class, and pass the username, and password as params to the method.  Because this could be used to generate the token for further scenarios as well.

Reply:
Updated and moved the function to util package and made it generic.